server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -610,10 +625,10 @@ private OpVsLuceneDocStatus compareOpToLuceneDocBasedOnSeqNo(final Operation op)
                     if (op.primaryTerm() > existingTerm) {
                         status = OpVsLuceneDocStatus.OP_NEWER;
                     } else {
-                        status = OpVsLuceneDocStatus.OP_STALE_OR_EQUAL;
```

Comment:
can you please add comments here in what cases we can get into these situations?